ping.go

```diff
@@ -0,0 +1,107 @@
+// Package ping provides a facility for pinging hosts that uses the installed ping command
+// rather than Go's ICMP support, enabling basic pinging without needing administrator/root
+// privileges.
+package ping
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"os/exec"
+	"strconv"
+
+	"github.com/getlantern/errors"
+)
+
+const (
+	// DefaultCount is 1
+	DefaultCount = 1
+
+	// DefaultPayloadSize is 1472 bytes
+	DefaultPayloadSize = 1472
+)
+
+// Opts specifies options for pinging
+type Opts struct {
+	// Count is the number of times to ping (defaults to 1)
+	Count int
+
+	// PayloadSize is the size of the data field in the ICMP packet, defaults to 1472 (for a 1500 byte total packet)
+	PayloadSize int
+}
+
+// Stats represents the statistics of pinging a host
+type Stats struct {
+	// RTT Minimum milliseconds
+	RTTMin float64
+	// RTT Average milliseconds
+	RTTAvg float64
+	// RTT Maximum milliseconds
+	RTTMax float64
+	// Packet Loss Rate (ratio, not percent)
+	PLR float64
+}
+
+// Run pings the specified host the specified number of times
+func Run(host string, opts *Opts) (*Stats, error) {
+	if opts == nil {
+		opts = &Opts{}
+	}
+	if opts.Count <= 0 {
+		opts.Count = DefaultCount
+	}
+	if opts.PayloadSize <= 0 {
+		opts.PayloadSize = DefaultPayloadSize
+	}
+
+	out, err := exec.Command("ping", args(host, opts)...).Output()
+	if err != nil {
+		return nil, errors.New("Unable to run ping command. Output: %v: %v", string(out), err)
```

Comment:
Nit: errors should start with a lowercase character.

Applies to all of the errors in this file.

Reply:
Oh yeah, I've been breaking that convention for so long now that it's hard to unlearn. Fixed.

Reply:
Haha

Thanks!